simpleini: make 'format' param more robust

BernhardDenner opened the pull request:
Build a more robust scanf format specifier out of the given 'format'
config setting. This allows formats like "%=%".

Additionally, this fixes a mem corruption vulnerability, due to unsafe
scanf format pattern.

# Checklist

Please only check relevant points.
For docu fixes, spell checking and similar nothing
needs to be checked.

- [x] commit messages are fine (with references to issues)
- [x] I ran all tests and everything went fine
- [ ] I added unit tests
- [x] affected documentation is fixed
- [x] I added code comments, logging, and assertions
- [x] meta data is updated (e.g. README.md of plugins)

@markus2330 please review my pull request

markus2330 commented on src/plugins/simpleini/simpleini.c:
```diff
@@ -18,33 +18,48 @@
 #include <kdbease.h>
 #include <kdberrors.h>
 #include <kdblogger.h>
+#include <kdbprivate.h>    // elektraStrNDup
```
It somehow seems wrong that elektraStrNDup is not in kdbhelpers.h.

BernhardDenner replied:
Yes, I was wondering why `elektraStrDup` is in kdbhelpers.h and `elektraStrNDup` is in kdbprivate.h

markus2330 replied:
Could you please move the forward declaration to `kdbhelpers.h` and avoid including kdbprivate?

markus2330 commented on src/plugins/simpleini/simpleini.c:
```diff
@@ -126,8 +249,17 @@ int elektraSimpleiniGet (Plugin * handle, KeySet * returned, Key * parentKey)
+	char * format = getReadFormat (handle);
+	if (!format)
+	{
+		ELEKTRA_SET_ERROR (ELEKTRA_ERROR_NOEOF, parentKey, "invalid 'format' specified");
```
Is NOEOF the correct error here?

BernhardDenner replied:
More or less. The description says "Invalid line encountered" which is some how related to this.
But actually I was too lazy to introduce a new error for this ;)
I'll change this.

markus2330 replied:
Maybe a general error for "invalid plugin configuration" is enough for such cases?

markus2330 replied:
Ideally it would already be validated by `checkConf`, see https://master.libelektra.org/doc/tutorials/plugins.md which is, however, currently not present in simpleini.

markus2330 commented on src/plugins/simpleini/simpleini.c:
```diff
@@ -126,8 +249,17 @@ int elektraSimpleiniGet (Plugin * handle, KeySet * returned, Key * parentKey)
+
 	FILE * fp = fopen (keyString (parentKey), "r");
 	if (!fp)
 	{
```
Shouldn't we free `format` here?

BernhardDenner commented:
- did a 'reformat-source'
- added some test cases, basically for reading and 'format' checks
- added some missings frees; ran valgrind and didn't find any errors
tests could be better, but for now this should be enough ;)

markus2330 commented:
Thank you! I fixed the issue with elektraStrNDup in 20f0aa1ee33f15aef8a6b0eb1ae671ef6a692978
